sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/ComponentSamples.cs

```diff
@@ -85,22 +86,19 @@ public async Task RunSamplesAsync(DigitalTwinsClient client)
             #region Snippet:DigitalTwinsSampleGetBasicDigitalTwin
 
             Response<BasicDigitalTwin> getBasicDtResponse = await client.GetDigitalTwinAsync<BasicDigitalTwin>(basicDtId);
-            if (getBasicDtResponse.GetRawResponse().Status == (int)HttpStatusCode.OK)
-            {
-                BasicDigitalTwin basicDt = getBasicDtResponse.Value;
-
-                // Must cast Component1 as a JsonElement and get its raw text in order to deserialize it as a dictionary
-                string component1RawText = ((JsonElement)basicDt.Contents["Component1"]).GetRawText();
-                IDictionary<string, object> component1 = JsonSerializer.Deserialize<IDictionary<string, object>>(component1RawText);
-
-                Console.WriteLine($"Retrieved and deserialized digital twin {basicDt.Id}:\n\t" +
-                    $"ETag: {basicDt.ETag}\n\t" +
-                    $"Prop1: {basicDt.Contents["Prop1"]}\n\t" +
-                    $"Prop2: {basicDt.Contents["Prop2"]}\n\t" +
-                    $"Component1 metadata: {component1[DigitalTwinsJsonPropertyNames.DigitalTwinMetadata]}\n\t" +
-                    $"Component1.Prop1: {component1["ComponentProp1"]}\n\t" +
-                    $"ComponentProp2: {component1["ComponentProp2"]}");
-            }
+            BasicDigitalTwin basicDt = getBasicDtResponse.Value;
+
+            // Must cast Component1 as a JsonElement and get its raw text in order to deserialize it as a dictionary
+            string component1RawText = ((JsonElement)basicDt.Contents["Component1"]).GetRawText();
+            var component1 = JsonSerializer.Deserialize<BasicDigitalTwinComponent>(component1RawText);
```

Comment:
Digital twins are custom objects defined by users using the Digital Twin Definition Language v2, using JSON-LD. A certain number of properties are required, like $dtid and $metadata, but depending on the model the twin adheres to, it may have additional properties. The user can create their own digital twin classes for serialization, but that requires some foreknowledge of the model. We offer BasicDigitalTwin as a way to explore unknown digital twins. The `JsonExtensionData` attribute dictionary helps deserialize those additional properties, but it has a constraint of being a `Dictionary<string, object>` (or value of JsonElement), so the experience isn't awesome.

So the better experience for a customer is to author their own custom digital twin POCOs. Absent of that, we at least provide this class for exploration.